transportable-udfs-trino-plugin/src/main/java/com/linkedin/transport/trino/TransportConnectorFactory.java

```diff
@@ -0,0 +1,99 @@
+import com.google.inject.Module;
+import io.airlift.log.Logger;
+import io.trino.spi.connector.Connector;
+import io.trino.spi.connector.ConnectorContext;
+import io.trino.spi.connector.ConnectorFactory;
+import io.trino.spi.function.FunctionId;
+import java.io.File;
+import java.io.FileFilter;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.ServiceLoader;
+import java.util.stream.Collectors;
+
+import static java.util.Objects.*;
+
+
+public class TransportConnectorFactory implements ConnectorFactory {
+
+  private static final String DEFAULT_TRANSPORT_UDF_REPO = "/transport-udf-repo";
+  private static final String TRANSPORT_UDF_REPO_CONFIG_NAME = "transport.udf.repo";
+  private static final Logger log = Logger.get(TransportConnectorFactory.class);
+
+  private static final FileFilter TRANSPORT_UDF_JAR_FILTER = (file) -> {
+    return file.isFile() && file.getName().endsWith(".jar") && !file.getName().startsWith("transportable-udfs");
+  };
+
+  private Connector connector;
+  private final Class<? extends Module> module;
+
+
+  // test only
+  public TransportConnectorFactory(Connector connector) {
+    this.connector = connector;
+    this.module = TransportModule.class;
+  }
+
+  public TransportConnectorFactory(Class<? extends Module> module) {
+    this.connector = null;
+    this.module = module;
+  }
+
+  @Override
+  public String getName() {
+    return "TRANSPORT";
+  }
+
+  @Override
+  public Connector create(String catalogName, Map<String, String> config, ConnectorContext context) {
+    requireNonNull(config, "config  is null");
+    if (this.connector == null) {
+      ClassLoader classLoaderForFactory = TransportConnectorFactory.class.getClassLoader();
+      List<URL> jarUrlList = getUDFJarUrls(config);
+      TransportUDFClassLoader classLoaderForUdf = new TransportUDFClassLoader(classLoaderForFactory, jarUrlList);
+      ServiceLoader<StdUdfWrapper> serviceLoader = ServiceLoader.load(StdUdfWrapper.class, classLoaderForUdf);
+      List<StdUdfWrapper> stdUdfWrappers = ImmutableList.copyOf(serviceLoader);
+      Map<FunctionId, StdUdfWrapper> functions = new HashMap<>();
+      for (StdUdfWrapper wrapper : stdUdfWrappers) {
+        log.info("Loading Transport UDF class: " + wrapper.getFunctionMetadata().getFunctionId().toString());
+        functions.put(wrapper.getFunctionMetadata().getFunctionId(), wrapper);
+      }
+      this.connector = new TransportConnector(functions);
+    }
+    return this.connector;
+  }
+
+  private List<URL> getUDFJarUrls(Map<String, String> config) {
+    String udfDir = config.containsKey(TRANSPORT_UDF_REPO_CONFIG_NAME)
+        ? config.get(TRANSPORT_UDF_REPO_CONFIG_NAME) : DEFAULT_TRANSPORT_UDF_REPO;
+    File[] udfSubDirs = new File(System.getProperty("user.dir") + udfDir).listFiles(File::isDirectory);
```

Comment:
why do we look for UDFs in a dir relative to the CWD? seems like we should allow for configuring an absolute path?

side note, to get CWD I think it's cleaner to use the `Path` API ([some examples on SO](https://stackoverflow.com/a/51028045/4979203))

Reply:
IMO we do not need to use CWD here. See Trino's [ServerPluginsProviderConfig](https://github.com/trinodb/trino/blob/9c2f86d941f2adebb8db94fdb3382b11f02f25ce/core/trino-main/src/main/java/io/trino/server/ServerPluginsProviderConfig.java#L22) when `plugin.dir` is not configured it use the relative `plugin/` by default.  The working directory when plugins are loaded is expected to have `etc/`, `plugin/` ... etc. 

@yiqiangin Can you move this config to a `TransportConfig` class ([comment](https://github.com/linkedin/transport/pull/128#discussion_r1175901967))?

```
public class TransportConfig
{
    private File transportUdfRepo = new File("transport-udf-default-path");

    public File getTransportUdfRepo()
    {
        return transportUdfRepo;
    }

    @Config("transport-udf-repo")
    public TransportConfig setTransportUdfRepo(File transportUdfRepo)
    {
        this.transportUdfRepo = transportUdfRepo;
        return this;
    }
}
```

Then for the constructor of `TransportConnector` you can use the repo from the config to load the function jar.

Reply:
Done.